### PR TITLE
Export `vec_unrep()`

### PR DESCRIPTION
**Pull request by DavisVaughan**
@lionel- do you like the name `vec_compress()` any more than `vec_unrep()`? It occurred to me that "compress" is pretty expressive of what this function does, but "unrep" fits the function family a bit more.

### CODE REVIEW

File: R/rep.R [hunk 1]
```diff
@@ -5,10 +5,22 @@
...
+#' These functions work along the size of `x`, rather than its length. For data
+#' frames, this means that rows are repeated rather than columns.
+#'
+#' Using `vec_unrep()` and `vec_rep_each()` together is very similar to using
+#' [base::rle()] and [base::inverse.rle()]. The following invariant shows
+#' the relationship between the two functions:
+#'
+#' ```
+#' compressed <- vec_unrep(x)
+#' x == vec_rep_each(compressed$key, compressed$times)
```
Comment (lionel-) [inline R/rep.R:22]: The `==` is a bit confusing because of the vectorised R operation. Perhaps `identical()` or `===`?

File: R/rep.R [hunk 2]
```diff
@@ -5,10 +5,22 @@
...
 #' @details
-#' `vec_rep()` and `vec_rep_each()` work along the size of `x`, rather than
-#' its length. For data frames, this means that rows are repeated rather
-#' than columns.
+#' These functions work along the size of `x`, rather than its length. For data
+#' frames, this means that rows are repeated rather than columns.
+#'
+#' Using `vec_unrep()` and `vec_rep_each()` together is very similar to using
+#' [base::rle()] and [base::inverse.rle()]. The following invariant shows
+#' the relationship between the two functions:
```
Comment (lionel-) [inline R/rep.R:18]: Add a boundary unit test for the empty case?

```r
compressed <- vec_unrep(integer())
expect_equal(
  vec_rep_each(compressed$key, compressed$times),
  integer()
)
```

**Review by lionel- — APPROVED** [review]
To me "unrep" is clearer than "compress" because it's more explicit about how the vector is compressed. What do you think @hadley?

File: R/rep.R [hunk 3]
```diff
@@ -5,10 +5,22 @@
...
+#'   repeated.
+#'
 #' @details
-#' `vec_rep()` and `vec_rep_each()` work along the size of `x`, rather than
-#' its length. For data frames, this means that rows are repeated rather
-#' than columns.
+#' These functions work along the size of `x`, rather than its length. For data
+#' frames, this means that rows are repeated rather than columns.
+#'
+#' Using `vec_unrep()` and `vec_rep_each()` together is very similar to using
```
Comment (hadley) [inline R/rep.R:16]: I'd add something here just being explicit about how it's different to rle. Also might worth drawing connection to `count()`/`uncount()`?

**Review by hadley — APPROVED** [review]
`vec_unrep()` seems like a decent name to me.
